service-base/src/main/java/org/eclipse/hono/service/credentials/CredentialsService.java

```diff
@@ -37,29 +36,50 @@
      *         given authID and type are registered for the tenant, the <em>status</em> will be {@link HttpURLConnection#HTTP_OK}
      *         and the <em>payload</em> will contain the details of the credentials.
      *         Otherwise the status will be {@link HttpURLConnection#HTTP_NOT_FOUND}.
+     * @throws NullPointerException if any of the parameters is {@code null}.
      */
     void getCredentials(String tenantId, String type, String authId, Handler<AsyncResult<CredentialsResult>> resultHandler);
 
-
-
     /**
      * Add credentials to credentials registry.
      *
-     * TODO: params
-     * 
+     * @param tenantId The tenant the device belongs to.
+     * @param credentialsObject A map containing keys and values that fulfill the credentials format of the credentials api.
+     *                  See <a href="https://www.eclipse.org/hono/api/Credentials-API/#credentials-format">Credentials Format</a> for details.
+     * @param resultHandler The handler to invoke with the result of the operation. If a device with the given ID does not
+     *         exist for the tenant, the <em>status</em> will be {@link HttpURLConnection#HTTP_PRECON_FAILED}.
+     *                      If credentials for the type and authId of the device are already existing, the <em>status</em> will be {@link HttpURLConnection#HTTP_CONFLICT}.
+     *         Otherwise the operation is successful and the status will be {@link HttpURLConnection#HTTP_CREATED}.
+     * @throws NullPointerException if any of the parameters is {@code null}.
```

Comment:
what about sending back a 400 Bad Request or throwing an `IllegalArgumentException` in case the given JSON does not contain all expected values?